server/src/test/java/org/apache/druid/server/coordinator/LoadQueuePeonTest.java

```diff
@@ -336,6 +336,12 @@ public void execute()
     Assert.assertTrue(timing.forWaiting().awaitLatch(segmentLoadedSignal));
     Assert.assertEquals(0, loadQueuePeon.getSegmentsToLoad().size());
     Assert.assertEquals(0L, loadQueuePeon.getLoadQueueSize());
+    curator.delete().guaranteed().forPath(loadRequestPath);
+    while (null != curator.checkExists().forPath(loadRequestPath)) {
+      Thread.sleep(5);
```

Comment:
Would it be better to have a separate countdown latch of size 2 and decrement it in the CuratorLoadQueuePeon callback and do an Assert with awaitLatch?

Reply:
Done